### PR TITLE
[exporter] Save compile configurations

### LINKED ISSUE #53: Add batch size and sequence length compile options to config
When compiling a model with `optimum-cli export neuron` it would be useful to store the compiled model batch size and sequence length values so that the inference server can use this information when creating the tokenizer for inference requests

### PR DESCRIPTION
**Pull request by JingyaHuang**
fix #53 
* Save static shapes during compilation into config.
* Save compilation arguments during compilation into config.
* Enhancement on the export arguments.

### CODE REVIEW

File: optimum/commands/export/neuron.py [hunk 1]
```diff
@@ -65,8 +65,8 @@ def parse_args_neuron(parser: "ArgumentParser"):
     )
     optional_group.add_argument(
         "--auto_cast_type",
-        default=None,
-        help='The data type to cast FP32 operations to when auto-cast mode is enabled. Can be `"bf16"`, `"fp16"` or `"tf32"`.',
+        default="bf16",
```
Comment (michaelbenayoun) [inline optimum/commands/export/neuron.py:68]: Maybe add a choices argument to make sure it is one of the 4 values.

File: optimum/commands/export/neuronx.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def parse_args_neuronx(parser: "ArgumentParser"):
     )
     optional_group.add_argument(
         "--auto_cast_type",
-        default=None,
+        default="bf16",
```
Comment (michaelbenayoun) [inline optimum/commands/export/neuronx.py:68]: Same comment

File: optimum/commands/export/neuron.py (hunk 1 above)
Comment (JingyaHuang) [reply optimum/commands/export/neuron.py:71]: Done!

File: optimum/commands/export/neuron.py [hunk 3]
```diff
@@ -60,12 +60,16 @@ def parse_args_neuron(parser: "ArgumentParser"):
     )
     optional_group.add_argument(
         "--auto_cast",
+        type=str,
         default=None,
+        choices=["None", "matmul", "all"],
```
Comment (michaelbenayoun) [inline optimum/commands/export/neuron.py:65]:
```suggestion
        choices=["none", "matmul", "all"],
```

Comment (JingyaHuang) [reply optimum/commands/export/neuron.py]: Corrected with changes in __main__ [here](https://github.com/huggingface/optimum-neuron/pull/55/commits/ff62fdef264728d1cddc339077e53863a0689035)

**Pull request merged**